sky/clouds/service_catalog/data_fetchers/fetch_gcp.py

```diff
@@ -1,579 +1,308 @@
-    }
-    df.rename(columns=column_remapping, inplace=True)
-
-    df = df[['AcceleratorName', 'Region', 'Price', 'SpotPrice']]
-    # Fix GPU names (i.e., remove NVIDIA prefix).
-    df['AcceleratorName'] = df['AcceleratorName'].apply(
-        lambda x: x.replace('NVIDIA ', ''))
-    # Add GPU counts.
-    df['AcceleratorCount'] = df['AcceleratorName'].apply(
-        lambda x: GPU_TYPES_TO_COUNTS[x])
-
-    # Parse the prices.
-    pattern = re.compile(r'\$?(.*?)\s?per GPU')
-
-    def parse_price(price_str: str) -> float:
-        if NOT_AVAILABLE_STR in price_str:
-            return float('nan')
-        try:
-            price = float(price_str[1:])
-        except ValueError:
-            price_match = re.search(pattern, price_str)
-            if price_match is None:
-                raise ValueError(f'Cannot parse price: {price_str}') from None
-            price = float(price_match.group(1))
-        return price
-
-    df['Price'] = df['Price'].apply(parse_price)
-    df['SpotPrice'] = df['SpotPrice'].apply(parse_price)
-
-    # Remove unsupported regions.
-    df = df[~df['Price'].isna()]
-    df = df[~df['SpotPrice'].isna()]
+            unit_price = _get_unit_price(sku)
+            gpu_price = unit_price * row['AcceleratorCount']
+            break
+
+        if gpu_price is not None:
+            return gpu_price
+
+        # Not found in the SKUs.
+        # FIXME(woosuk): Remove this once the SKUs are updated.
+        gpu = row['AcceleratorName']
+        region = row['Region']
+        if gpu == 'T4':
+            assert region in ['europe-west1', 'europe-west2', 'asia-east2'], row
+        elif gpu == 'P100':
+            assert region in ['europe-west4', 'australia-southeast1'], row
+        return None
+
+    df['Price'] = df.apply(lambda row: get_gpu_price(row, spot=False), axis=1)
+    df['SpotPrice'] = df.apply(lambda row: get_gpu_price(row, spot=True),
+                               axis=1)
+    # Drop invalid rows.
+    df = df[df['Price'].notna() & df['SpotPrice'].notna()]
+    df = df.reset_index(drop=True)
     return df
 
 
-def get_gpu_zones(url) -> pd.DataFrame:
-    page = requests.get(url)
-    df = pd.read_html(page.text.replace('<br>', '\n'))[0]
-    column_remapping = {
-        'GPU platforms': 'AcceleratorName',
-        'Zones': 'AvailabilityZone',
-    }
-    df.rename(columns=column_remapping, inplace=True)
-    df = df[['AvailabilityZone', 'AcceleratorName']]
+def get_tpu_df(skus: List[Dict[str, Any]]) -> pd.DataFrame:
+    df = pd.read_csv(GCP_METADATA_URL + 'tpus.csv')
+
+    def get_tpu_price(row: pd.Series, spot: bool) -> float:
+        tpu_price = None
+        for sku in skus:
+            # NOTE: Because us-east1-d is a hidden zone that supports TPUs,
+            # no price information is available for it. As a workaround,
+            # we asssume that the price is the same as us-central1.
+            tpu_region = row['Region']
+            if tpu_region == 'us-east1':
```

Comment:
Good point. The two regions `us-east1` and `us-west1` do not appear in the GCP [webpage](https://cloud.google.com/tpu/pricing). Moreover, they do not appear in the SKU page. This is why I think they are "hidden".

![Screen Shot 2023-02-14 at 12 34 03 PM](https://user-images.githubusercontent.com/46394894/218855991-59894258-6c55-4eff-8e50-61d9bbc4c9d4.png)

Reply:
Surprisingly, even in the billing API, these two regions do not appear for `Tpu-v3 Preemptible Pod Accelerator USA`. I'm not sure this is because the regions do not actually support spot TPU pods.